api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -295,7 +297,14 @@ export class ScenariosService {
     if (isRight(isLegacyProjectCompleted) && !isLegacyProjectCompleted.right)
       return left(projectNotReady);
 
-    const scenario = await this.crudService.create(validatedMetadata, info);
+    let scenario;
+    try {
+      scenario = await this.crudService.create(validatedMetadata, info);
+      scenario = await this.crudService.getById(scenario.id);
+    } catch (e) {
+      return left(scenarioNotCreated);
+    }
```

Comment:
Very minor, but... I am not sure about the rationale for this: why do we expect to handle exceptions when creating the scenario in a more surgical way, so to speak, than before?

In case the creation fails (it's the 1000th scenario in the project, for example), then `create()` should fail outright and this would trigger the global exception filter in the previous implementation where we were not using a `try` block... I think that should be enough (especially as we don't provide additional information about the _why_ of a possible failure here - our error message says that the scenario could not be created within the given project, but we don't give a reason (not saying we should - even discriminating between violation of different constraints, from `pg` errors, is kinda clunky and overkill IMO)

Also, shouldn't we get the scenario back from the call to `create()`, without having to `getById()` separately?

I think we can let things fail here, and as long as we do have a valid scenario returned by `create()` then the rest of the method can continue (otherwise, if `scenario` is undefined we should `return left(scenario)` or if an exception is thrown, either catch it - as your implementation does - or just let it bubble up).

Reply:
Yeah, I don't like it honestly, but for whatever reason, the create method wasn't returning the full entity, and in the response I would always get a projectScenarioId undefined. This would make subsequent tests that check for projectScenarioId fail.

Maybe it's not that big of a deal? Maybe instead of checking the response from the create endpoint, I could simply GET the scenario, and check the projectScenarioId there, but it still irks me that I couldn't get the full info. I left it like this because I thought that if it fails on the creation because of the MAX 999 id, it would be nicer to receive a more user friendly message instead of a 500.

The save method not returning the full entity seems like it was a bug bug/missing feature/whatever in typeorm, but it doesn't look like it ever got resolved
https://github.com/typeorm/typeorm/pull/5680

Reply:
ok, yes - I understand your rationale for this now, thanks.

The discussion past this comment in this thread is particularly insightful: https://github.com/typeorm/typeorm/issues/3490#issuecomment-585630457

Basically db `default`s and values set via trigger would not be set.

We could do what TypeORM doesn't do by implementing `BaseService.extendCreateResult()` for `ScenariosService`, something like

```
  async extendCreateResult(entity: Scenario, _createModel: CreateScenarioDTO): Promise<Scenario> {
    return await this.crudService.getById(entity.id);
  }
```

but pragmatically, it's just moving the `getById` in a different place, so I'm fully ok with keeping your implementation - maybe add a comment to clarify that we `create()` and then `getById()` in order to have the full entity back because some properties are set via trigger or defaults, and let's move on with life 😀